app/models/user_mentee_application.rb

```diff
@@ -29,6 +29,9 @@
 class UserMenteeApplication < ApplicationRecord
   belongs_to :user
   has_many :mentee_application_states, dependent: :destroy
+  # rubocop:disable Rails/InverseOf
+  has_one :current_state, -> { order(created_at: :desc).limit(1) }, class_name: 'MenteeApplicationState', dependent: nil
```

Comment:
shouldn't this order by `status`?? (technically we can do `created_at`)

Reply:
yes, it seems like both work (TIL). But order by status seems more readable to me

Reply:
@toyhammered yeah, ordered by `status` is more descriptive. Again, see comment about whipping it up.